src/lune/builtins/datetime/mod.rs

```diff
@@ -0,0 +1,222 @@
+                    fmt_str.to_string_lossy(),
+                    locale.to_string_lossy(),
+                ).map_err(|()| LuaError::external(
+                    "failed to parse DateTime object, invalid",
+                )))
+            },
+        )?
+        .build_readonly()
+}
+
+impl<'lua> FromLua<'lua> for TimestampType {
+    fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> LuaResult<Self> {
+        match value {
+            LuaValue::Integer(_) => Ok(TimestampType::Seconds),
+            LuaValue::Number(num) => Ok(if num.fract() == 0.0 {
+                TimestampType::Seconds
+            } else {
+                TimestampType::Millis
+            }),
+            _ => Err(LuaError::external(
+                "Invalid enum type, number or integer expected",
+            )),
+        }
+    }
+}
+
+impl LuaUserData for DateTime {
+    fn add_fields<'lua, F: LuaUserDataFields<'lua, Self>>(fields: &mut F) {
+        fields.add_field_method_get("unixTimestamp", |_, this| Ok(this.unix_timestamp));
+        fields.add_field_method_get("unixTimestampMillis", |_, this| {
+            Ok(this.unix_timestamp_millis)
+        });
+
+        fields.add_field_method_set("unixTimestamp", |_, this, val| {
+            this.unix_timestamp = val;
+
+            Ok(())
+        });
+
+        fields.add_field_method_set("unixTimestampMillis", |_, this, val| {
+            this.unix_timestamp_millis = val;
+
+            Ok(())
+        });
+    }
+
+    fn add_methods<'lua, M: LuaUserDataMethods<'lua, Self>>(methods: &mut M) {
+        methods.add_method("toIsoDate", |_, this, ()| {
+            Ok(this
+                .to_iso_date()
+                .map_err(|()| LuaError::external("failed to parse DateTime object, invalid")))
+        });
+
+        methods.add_method(
+            "formatTime",
+            |_, this, (timezone, fmt_str, locale): (LuaValue, LuaString, LuaString)| {
+                Ok(this
+                    .format_time(
+                        Timezone::from_lua(timezone, &Lua::new())?,
+                        fmt_str.to_string_lossy(),
+                        locale.to_string_lossy(),
+                    )
+                    .map_err(|()| LuaError::external("failed to parse DateTime object, invalid")))
+            },
+        );
+
+        methods.add_method("toUniversalTime", |_, this: &DateTime, ()| {
+            Ok(this.to_universal_time())
+        });
+
+        methods.add_method("toLocalTime", |_, this: &DateTime, ()| {
+            Ok(this.to_local_time())
+        });
+    }
+}
+
+impl<'lua> FromLua<'lua> for DateTime {
+    fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> LuaResult<Self> {
+        match value {
+            LuaValue::Nil => panic!("found nil"),
+            LuaValue::Table(t) => Ok(DateTime::from_unix_timestamp(
+                TimestampType::Seconds,
+                t.get("unixTimestamp")?,
+            )),
+            _ => panic!("invalid type"),
+        }
+    }
+}
```

Comment:
Where is this being used?

Reply:
Unsure what I declared this for, lol...

Reply:
Okay, I remember what I declared it for:

```lua
local DateTime = require("@lune/datetime")

DateTime.toIsoDate({
   unixTimestamp = 1693811157
})

-- instead of
DateTime.fromUnixTimestamp(1693811157):toIsoDate()
```